### PR TITLE
feat: support Stacks from Hist

### PR DESCRIPTION
**Pull request by henryiii**
Adds support for Stacks from Hist, as well as cleaning up a few things related to testing. Tests should have some expected output that they compare against. Also, the output test should be added to nox; GHA code should remain simple (just a suggestion). Again, the output should be checked against truth.

### CODE REVIEW

**Comment by ast0815** [comment]
I agree that it would be nice to have actual testable outputs for all tests, but it is a bit difficult to define the correct output of something that should just look like a correct histogram. The way you did it here, by comparing two different ways that are supposed to generate exactly the same output, is fine for test exactly that. But i think it gets a bit murkier when trying to define how the correct output should look like in the first place.

One could define a standard input and output and make the test fail when the generated one is not exactly the same. But every time something does change, one would still have to look at it and decide whether it is correct, and then update the standard output, which is a bit of a faff.

That is why the tests were set up to actually print out the output via the `-s` option. Are you suggesting we remove the ability to check the output visually in the GitHub tests? Because I would rather re-add `-s` option. I have never used `nox`, but I assume it is only used to run the tests locally.

File: .github/workflows/pythontests.yml [hunk 1]
```diff
@@ -20,20 +20,15 @@ jobs:
...
-        python -m pip install -e .[test]
+      run: python -m pip install -e .[test,boost,root]
     - name: Install optional dependencies for tests
-      run: |
-        pip install boost-histogram uproot awkward pandas || true
+      run: pip install pandas
     - name: Run tests
-      run: |
-        python -m pytest -s tests/test.py
+      run: python -m pytest
```
Comment (henryiii) [inline .github/workflows/pythontests.yml:31]:
```suggestion
      run: python -m pytest -s
```

File: noxfile.py [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    session.run("pre-commit", "run", "--all-files", *session.posargs)
+
+
+@nox.session(python=ALL_PYTHONS, reuse_venv=True)
+def tests(session):
+    """
+    Run the unit and regular tests.
+    """
+    session.install(".[test,boost,root]")
+    session.run("pytest", *session.posargs)
```
Comment (henryiii) [inline noxfile.py:25]:
```suggestion
    session.run("pytest", "-s", *session.posargs)
```

**Comment by henryiii** [comment]
I don't normally go look at the logs, so if I make a change, I expect the tests to tell me if something changed. Most of the time, you are likely refactoring, and explicitly not wanting the output to change, so you should be notified if it does. If you just trust looking at visual output, something might be completely wrong but not visually be obvious. Having checked output makes refactoring easier, not harder, because you can trust that nothing changed. The rare cases where you really are changing the output, you have to run it with -s and re-capture the output.

You can still output visual logs if you also check the logs.

Anyway, none of that was supposed to change here, I was just primarily adding Stacks support, and making it easier to run locally (with nox) so I could run the tests, well, locally. :)

**Comment by ast0815** [comment]
Sure. Will do one now.

**Comment by ast0815** [comment]
Done.

**Pull request merged**